airflow/kubernetes/pod_generator.py

```diff
@@ -324,47 +325,36 @@ def add_sidecar(pod: k8s.V1Pod) -> k8s.V1Pod:
     @staticmethod
     def from_obj(obj) -> Optional[k8s.V1Pod]:
         """Converts to pod from obj"""
+
         if obj is None:
             return None
 
-        if isinstance(obj, PodGenerator):
-            return obj.gen_pod()
-
-        if not isinstance(obj, dict):
-            raise TypeError(
-                'Cannot convert a non-dictionary or non-PodGenerator '
-                'object into a KubernetesExecutorConfig')
+        k8s_legacy_object = obj.get("KubernetesExecutor", None)
```

Comment:
The only thing that would have existed in dict under `KubernetesExecutor` is the `executor_config`, right?

Just double-checking to verify that we don't raise deprecation warning for other cases

Reply:
@kaxil yes, if the user uses the "KubernetesExecutor" key for the dict it will work exactly like the legacy system.